src/coreclr/jit/importer.cpp

```diff
@@ -7046,6 +7049,7 @@ void Compiler::impImportNewObjArray(CORINFO_RESOLVED_TOKEN* pResolvedToken, CORI
         }
```

Comment:
We may also want to add code to `LoadDynamicInfoEntry` / `case ENCODE_READYTORUN_HELPER` that rejects R2R images that contain calls to the broken `CORINFO_HELP_NEW_MDARR`. Otherwise, trying to run the broken code in these R2R images may cause existing apps to crash after installing this update.

Reply:
Excellent point, thanks @jkotas!
I will try to add this today.

Reply:
I have added a commit (https://github.com/dotnet/runtime/pull/64104/commits/295b0095fb38478e4fa2f20a91432d46d196a3a7) that implements just that. Thanks again for the suggestion @jkotas!